app/controllers/api/messaging.php

```diff
@@ -1744,21 +1914,212 @@
+        }
+
+        $data = $message->getAttribute('data');
+
+        if (!empty($content)) {
+            $data['content'] = $content;
         }
 
         $message->setAttribute('data', $data);
-        $message->setAttribute('search', $message->getId() . ' ' . $data['description'] . ' ' . $data['subject']);
+        $message->setAttribute('search', $message->getId() . ' ' . $data['description'] . ' ' . $message->getAttribute('providerId'));
 
         if (!empty($status)) {
             $message->setAttribute('status', $status);
         }
 
+        if (!empty($description)) {
+            $message->setAttribute('description', $description);
+        }
+
+        $message = $dbForProject->updateDocument('messages', $message->getId(), $message);
+
+        if ($status === 'processing') {
+            $messaging
+                ->setMessageId($message->getId())
+                ->setProject($project);
+
+            if (!empty($deliveryTime)) {
+                $messaging
+                ->setDeliveryTime($deliveryTime)
+                ->schedule();
+            } else {
+                $messaging->trigger();
+            }
+        }
+
+        $response
+            ->dynamic($message, Response::MODEL_MESSAGE);
+    });
+
+App::patch('/v1/messaging/messages/push/:messageId')
+    ->desc('Update a push notification.')
+    ->groups(['api', 'messaging'])
+    ->label('audits.event', 'messages.update')
+    ->label('audits.resource', 'messages/{response.$id}')
+    ->label('scope', 'messages.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN, APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'messaging')
+    ->label('sdk.method', 'updatePushNotification')
+    ->label('sdk.description', '/docs/references/messaging/update-push-notification.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_MESSAGE)
+    ->param('messageId', '', new UID(), 'Message ID.')
+    ->param('to', [], new ArrayList(new Text(Database::LENGTH_KEY)), 'List of Topic IDs or List of User IDs or List of Target IDs.', true)
+    ->param('description', '', new Text(256), 'Description for Message.', true)
+    ->param('title', '', new Text(256), 'Title for push notification.', true)
+    ->param('body', '', new Text(64230), 'Body for push notification.', true)
+    ->param('data', null, new JSON(), 'Additional Data for push notification.', true)
+    ->param('action', '', new Text(256), 'Action for push notification.', true)
+    ->param('icon', '', new Text(256), 'Icon for push notification.', true)
+    ->param('sound', '', new Text(256), 'Sound for push notification.', true)
+    ->param('color', '', new Text(256), 'Color for push notification.', true)
+    ->param('tag', '', new Text(256), 'Tag for push notification.', true)
+    ->param('badge', '', new Text(256), 'Badge for push notification.', true)
```

Comment:
Let's include in the descriptions which platforms can use which of these,f or example `badge` is Apple only. The info should be included in the messaging lib docs

Reply:
done this